File: abstractions/java/lib/src/main/java/com/microsoft/kiota/store/BackingStoreFactory.java
package com.microsoft.kiota.store;

import javax.annotation.Nonnull;

/** Defines the contract for a factory that creates backing stores. */
public interface BackingStoreFactory {
    /** 
     * Creates a new instance of the backing store.
     * @return a new instance of the backing store.
     */
    @Nonnull
    BackingStore createBackingStore();
}

File: abstractions/php/src/Store/BackingStoreFactory.php
<?php

namespace Microsoft\Kiota\Abstractions\Store;

/** Defines the contract for a factory that creates backing stores. */
interface BackingStoreFactory
{
    /**
     * Creates a new instance of the backing store.
     * @return BackingStore a new instance of the backing store.
     */
    public function createBackingStore(): BackingStore;
}